### ISSUE TITLE
Endless Loop on SLO

### ISSUE THREAD

**Issue by tflori** [codegreencreative/laravel-samlidp#36]
I've setup the logout to logout after slo (`config(samlidp.logout_after_slo)`) and because no service provider has a valid setup with a logout procedure the session also does not save a variable `saml.slo` (see https://github.com/codegreencreative/laravel-samlidp/blob/master/src/Http/Controllers/LogoutController.php#L30). And in return the logout redirects to the `/saml/logout` url again (see https://github.com/codegreencreative/laravel-samlidp/blob/master/src/Listeners/SamlLogout.php#L19).

Basically there are two solutions for that:
1. save an empty array in the session if there is none (in the LogoutController)
2. check differently if we are in the saml logout process (for example check the request url)

I can create a PR if you tell me which solution you prefer. My preference is to set the session variable if not set already.

**Comment by upwebdesign** [comment]
@tflori Let's see what you do with the session idea. I am able to replicate the same issue you are experiencing. I will be curious how you implement your changes!

**Comment by tflori** [comment]
@upwebdesign I've created two pull requests as we would need it in 2.x

Thanks in advance

**Comment by tflori** [comment]
thx, it works now

**Comment by waqleh** [comment]
after updating from version ^2.0 to version ^3.0 this issue is back! @upwebdesign can you apply this fix to version 3+ also

**Comment by upwebdesign** [comment]
@waqleh Ive updated and released v3.2.3. Give it a try and let me know if you find any issues.

**Comment by waqleh** [comment]
I had to add `'guards' => ['web'],` to `config/samlidp.php` but it works now, thanks

**Comment by waqleh** [comment]
@upwebdesign this fix is missing from version 4, can it be merged to v4 please?

**Comment by waqleh** [comment]
@upwebdesign I created a merge request for it https://github.com/codegreencreative/laravel-samlidp/pull/73

**Comment by upwebdesign** [comment]
@waqleh Thank you for this. I will try to get this into v4 with a new version as soon as I can.
